projects/kubernetes/kubernetes/kubelet.sh

```diff
@@ -2,6 +2,15 @@
 # Kubelet outputs only to stderr, so arrange for everything we do to go there too
 exec 1>&2
 
+if [ -f /var/config/kubelet/disabled ] ; then
+    echo "kubelet.sh: /var/config/kubelet/disabled file is present, exiting"
+    exit 0
+fi
+if ! [ -z "$KUBELET_DISABLED" ] ; then
```

Comment:
`if ! [ -z "FOO" ]` is the same as `if [ -n "FOO" ]` and I think the latter is clearer.

I think this all needs to go after `. /etc/kubelet.sh.conf` since that might be how one wants to set `KUBELET_DISABLED`.

That in turn seems to suggest the `. /etc/kubelet.sh.conf` should be moved above the CNI unpack, since if we are exiting we want to skip that too.

So the order should be:
```
if [ -e /etc/kubelet.sh.conf ] ; then
    . /etc/kubelet.sh.conf
fi

<both blocks of your new code>

if [ ! -e /var/lib/cni/.opt.defaults-extracted ] ; then
    mkdir -p /var/lib/cni/opt/bin
    tar -xzf /root/cni.tgz -C /var/lib/cni/opt/bin
    touch /var/lib/cni/.opt.defaults-extracted
fi
```